support/k3d/k3d.go

```diff
@@ -0,0 +1,399 @@
+		args = append(args, "--image", c.image)
+	}
+
+	args = append(args, c.args...)
+	cmd := fmt.Sprintf("%s cluster create %s", c.path, c.name)
+
+	if len(args) > 0 {
+		cmd = fmt.Sprintf("%s %s", cmd, strings.Join(args, " "))
+	}
+	log.V(4).InfoS("Launching k3d cluster", "command", cmd)
+
+	var stdout, stderr bytes.Buffer
+	p := utils.RunCommandWithCustomWriter(cmd, &stdout, &stderr)
+	if p.Err() != nil {
+		return "", fmt.Errorf("k3d: failed to create cluster %q: %s: %s: %s %s", c.name, p.Err(), p.Result(), stdout.String(), stderr.String())
+	}
+	clusters, ok := c.clusterExists(c.name)
+	if !ok {
+		return "", fmt.Errorf("k3d cluster create: cluster %v still not in 'cluster list' after creation: %v", c.name, clusters)
+	}
+	log.V(4).Info("k3d clusters available: ", clusters)
+
+	kConfig, err := c.getKubeConfig()
+	if err != nil {
+		return "", err
+	}
+	return kConfig, c.initKubernetesAccessClients()
+}
+
+func (c *Cluster) CreateWithConfig(ctx context.Context, configFile string) (string, error) {
+	var args []string
+	if configFile != "" {
+		args = append(args, "--config", configFile)
+	}
+	return c.Create(ctx, args...)
+}
+
+func (c *Cluster) GetKubeconfig() string {
+	return c.kubeConfigFile
+}
+
+func (c *Cluster) GetKubectlContext() string {
+	return fmt.Sprintf("k3d-%s", c.name)
+}
+
+func (c *Cluster) ExportLogs(ctx context.Context, dest string) error {
+	log.Warning("ExportLogs not implemented for k3d. Please use regular kubectl like commands to extract the logs from the cluster")
+	return nil
+}
+
+func (c *Cluster) Destroy(ctx context.Context) error {
+	log.V(4).InfoS("Destroying k3d cluster", "name", c.name)
+	if err := c.findOrInstallK3D(); err != nil {
+		return fmt.Errorf("failed to find or install k3d: %w", err)
+	}
+
+	if _, ok := c.clusterExists(c.name); !ok {
+		log.V(4).InfoS("Skipping k3d cluster destruction. Cluster does not exist", "name", c.name)
+		return nil
+	}
+
+	cmd := fmt.Sprintf("%s cluster delete %s", c.path, c.name)
+	log.V(4).InfoS("Destroying k3d cluster", "command", cmd)
+	p := utils.RunCommand(cmd)
+	if p.Err() != nil {
+		outBytes, err := io.ReadAll(p.Out())
+		if err != nil {
+			log.ErrorS(err, "failed to read data from the k3d cluster delete process output due to an error")
+		}
+		return fmt.Errorf("k3d: failed to delete cluster %q: %s: %s: %s", c.name, p.Err(), p.Result(), string(outBytes))
+	}
+
+	log.V(4).InfoS("Removing kubeconfig file", "configFile", c.kubeConfigFile)
+	if err := os.RemoveAll(c.kubeConfigFile); err != nil {
+		return fmt.Errorf("k3d: failed to remove kubeconfig file %q: %w", c.kubeConfigFile, err)
+	}
+	return nil
+}
+
+func (c *Cluster) SetDefaults() support.E2EClusterProvider {
+	if c.path == "" {
+		c.path = "k3d"
+	}
+	return c
+}
+
+func (c *Cluster) WaitForControlPlane(ctx context.Context, client klient.Client) error {
+	log.V(4).Info("k3d doesn't implement a WaitForControlPlane handler. The --wait argument passed to the `k3d` should take care of this already")
```

Comment:
I would reword this and just say that "the k3d provider doesn't implement a WaitForControlPlane as the provider automatically wait for the control plane"

Reply:
One more question, what --wait argument ?
Are you saying that you can pass a --wait argument during cluster create ?

Reply:
Yes.. K3d has an arg `--wait` with cluster create.. Which also defaults to true anyway.. Let me reword the message as you suggested.